libappimage/test/CMakeLists.txt

```diff
@@ -1,28 +1,52 @@
 # Locate GTest
-find_package(GTest REQUIRED)
-include_directories(${GTEST_INCLUDE_DIRS})
-
-# Link runTests with what we want to test and the GTest and pthread library
-add_executable(test_appimagelib
-    test_appimagelib.cpp
-)
-target_compile_definitions(test_appimagelib
-    PRIVATE TEST_DATA_DIR="${CMAKE_CURRENT_SOURCE_DIR}/data/"
-)
-
-target_link_libraries(test_appimagelib 
-    appimage
-    pthread  
-    ${GTEST_LIBRARIES}
-    ssl
-    crypto
-    lzma
-    pthread
-    ${GLIB_LIBRARIES}
-    ${GIO_LIBRARIES}
-    ${CAIRO_LIBRARIES}
-)
-
-add_dependencies(test_appimagelib appimage)
-
-add_test(test_appimagelib test_appimagelib)
+
+enable_testing()
+
+if(BUILD_TESTING)
+    set(USE_SYSTEM_GTEST OFF CACHE BOOL "Use system GTest instead of downloading and building GTest")
+
+    if(NOT USE_SYSTEM_GTEST)
+        ExternalProject_Add(gtest
+            GIT_REPOSITORY https://github.com/google/googletest.git
```

Comment:
Not terribly happy about having yet another external dependency which we must build ourselves... can't we stick to something that comes with CentOS 6? cc @azubieta

Reply:
I asked before going with Google Test. What about glib test https://developer.gnome.org/glib/stable/glib-Testing.html

Reply:
Google test is actually the least annoying dependency we have. If you want to, we can install an up to date version in our Docker images in https://github.com/TheAssassin/AppImageBuild, but that won't really change anything. Google Test is one of the best unit test systems available for C++ at the moment, easy to use, works fine, and shows all the necessary information on test failures, which makes remote debugging Travis build failures a lot easier.

Reply:
In general, please try to use only such dependencies that are part of the CentOS 6 distribution. I am fine with Google Test if a) it is optional in the sense that if you don't have it the tests are simply skipped, and b) if we don't build it each time

Reply:
It is only built for the tests, and in large parts header-only. Check it out here: https://github.com/google/googletest/tree/master/googletest/src